app/assets/javascripts/infinity_scroll.js

```diff
@@ -0,0 +1,64 @@
+"use strict";
+function init_infinity_scroll(initial_page_number, request_url, load_first_page) {
+    var page_number;
+    if (initial_page_number === undefined) {
+        page_number = 1;
+    }
+    else {
+        page_number = initial_page_number
+    }
+    if (load_first_page === true) load_data();
+    start_infinity_scroll();
+
+    function start_infinity_scroll() {
+        $(window).on('scroll', function () {
+            if (end_of_page()) {
+                load_data();
+                $(window).off('scroll');
+            }
+        });
+    }
+
+    function end_of_page() {
+        return $(window).scrollTop() + $(window).height() > $(document).height() - 100
+    }
+
+    function load_data() {
```

Comment:
I don't understand issue with polymorphic and  comments "page" ... 
if you want to return only specific objects you can use "where(commentbale_type = 'Event') ... or what you need. Pages are not visible, they are used in JS only.
Also, you have working examples with gems. I don't want to ask to rework all this. Just try next time speak with mentor if something not works or you have troubles. Rule is = less code = less bugs